src/dynamic_batch_scheduler.cc

```diff
@@ -601,31 +611,34 @@ DynamicBatchScheduler::DelegateResponse(
   std::lock_guard<std::mutex> lock(completion_queue_mtx_);
   completion_queue_.emplace_back();
   auto queue_slot = &completion_queue_.back();
-  // Pass raw ptr to lambda for tracking stats from cache and updating
-  // metric reporter on cache miss stats after insertion
-  InferenceRequest* raw_request_ptr = request.get();
+  const auto key = request->CacheKey();
 
   request->SetResponseDelegator(
-      [this, queue_slot, raw_request_ptr](
+      [this, queue_slot, key](
           std::unique_ptr<InferenceResponse>&& response, const uint32_t flags) {
-        if (response_cache_enabled_ && raw_request_ptr->CacheKeyIsSet()) {
+        if (response_cache_enabled_) {
           // Cache insertion happens here because we need the backend to have
           // computed the inference response first in the case of cache miss
-          auto cache = model_->Server()->GetResponseCache();
-          auto status = cache->Insert(*response, raw_request_ptr);
+          auto cache = model_->Server()->CacheManager()->Cache();
+          auto insert_start_ns = CaptureTimeNs();
+          auto status = cache->Insert(response.get(), key);
```

Comment:
I don't see handling for "holding" responses for decoupled model, is it going to be in a separate PR?

Reply:
No decoupled cache support before or after this PR, this is parity with existing cache, just moving to use of C API instead.